File: typescript/triangle/triangle.ts
const strictEqualSides = (x: readonly number[]): number => new Set(x).size;

const sum = (x: readonly number[]): boolean => {
  const [a, b, c] = x.slice().sort((a, b) => a - b);

  return a + b > c;
};

const checkSides = (x: readonly number[]): boolean =>
  x.some((side) => side <= 0);

const isTriangle = (x: readonly number[]): boolean => !checkSides(x) && sum(x);

export const whichTriangle = (x: readonly number[]): string => {
  if (!isTriangle(x)) return "That's not a triangle!";

  const equalSides = strictEqualSides(x);

  if (equalSides === 1) return 'equilateral';
  else if (equalSides === 2) return 'isosceles';
  else if (equalSides === 3) return 'scalene';
  else return "That's not a triangle!";
};

File: julia/triangle/triangle.jl
strict_equal_sides = x -> length(Set(x))

sum = x -> begin a, b, c = sort(x); a + b > c end

check_sides = x -> count(>=(0), x) < 3 ? false : true

is_triangle = x -> check_sides(x) && sum(x)

function is_equilateral(sides)
    if !is_triangle(sides)
        return false
    end

    if strict_equal_sides(sides) == 1
        return true
    else
        return false
    end
end

function is_isosceles(sides)
    if !is_triangle(sides)
        return false
    end

    res = strict_equal_sides(sides)

    if res == 2 || res == 1
        return true
    else
        return false
    end
end

function is_scalene(sides)
    if !is_triangle(sides)
        return false
    end

    if strict_equal_sides(sides) == 3
        return true
    else
        return false
    end
end
